scripts/download-rc.js

```diff
@@ -0,0 +1,152 @@
+    } else {
+        console.log(`Downloading: hazelcast enterprise test jar com.hazelcast:hazelcast-enterprise:${HAZELCAST_TEST_VERSION}:jar:tests`)
+        const subprocess = spawnSync('mvn', ['-q', 'org.apache.maven.plugins:maven-dependency-plugin:2.8:get', `-DrepoUrl=${ENTERPRISE_TEST_REPO}`, `-Dartifact=com.hazelcast:hazelcast-enterprise:${HAZELCAST_TEST_VERSION}:jar:tests`, `-Ddest=hazelcast-enterprise-${HAZELCAST_TEST_VERSION}-tests.jar`], {
+            stdio: 'inherit',
+            shell: ON_WINDOWS
+        })
+        if (subprocess.status !== 0) {
+            console.log(`Failed to download hazelcast enterprise test jar com.hazelcast:hazelcast-enterprise:${HAZELCAST_TEST_VERSION}:jar:tests`)
+            process.exit(1)
+        }
+    }
+    CLASSPATH = `hazelcast-enterprise-${HAZELCAST_ENTERPRISE_VERSION}.jar${ENV_VARIABLE_SEPERATOR}hazelcast-enterprise-${HAZELCAST_ENTERPRISE_TEST_VERSION}-tests.jar${ENV_VARIABLE_SEPERATOR}${CLASSPATH}`
+    console.log("Starting Remote Controller ... enterprise ...")
+} else {
+    if (fs.existsSync(`hazelcast-${HAZELCAST_VERSION}.jar`)) {
+        console.log("hazelcast.jar already exists, not downloading from maven.")
+    } else {
+        console.log(`Downloading: hazelcast jar com.hazelcast:hazelcast:${HAZELCAST_VERSION}`)
+        const subprocess = spawnSync('mvn', ['-q', 'dependency:get', `-DrepoUrl=${REPO}`, `-Dartifact=com.hazelcast:hazelcast:${HAZELCAST_VERSION}`, `-Ddest=hazelcast-${HAZELCAST_VERSION}.jar`], {
+            stdio: 'inherit',
+            shell: ON_WINDOWS
+        })
+        if (subprocess.status !== 0) {
+            console.log(`Failed download hazelcast jar com.hazelcast:hazelcast:${HAZELCAST_VERSION}`)
+            process.exit(1)
+        }
+    }
+    CLASSPATH = `hazelcast-${HAZELCAST_VERSION}.jar${ENV_VARIABLE_SEPERATOR}${CLASSPATH}`
+    console.log("Starting Remote Controller ... oss ...")
+}
+
+let sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+
+async function start() {
+    if (ON_WINDOWS) {
+        spawnSync(`start /min "hazelcast-remote-controller" cmd /c "java -Dhazelcast.enterprise.license.key=${HAZELCAST_ENTERPRISE_KEY} -cp ${CLASSPATH} com.hazelcast.remotecontroller.Main --use-simple-server> rc_stdout.txt 2>rc_stderr.txt"`, [], {
+            stdio: 'inherit',
+            shell: true
+        })
+
+        console.log("wait for Hazelcast to start ...")
+
+        spawnSync("ping -n 15 127.0.0.1 > nul", [], {
+            stdio: 'inherit',
+            shell: true
+        })
+    } else {
+        spawnSync(`nohup java -Dhazelcast.enterprise.license.key=${HAZELCAST_ENTERPRISE_KEY} -cp ${CLASSPATH} com.hazelcast.remotecontroller.Main --use-simple-server > rc_stdout.log 2>rc_stderr.log &`, [], {
+            stdio: 'inherit',
+            shell: true
+        })
+        console.log("wait for Hazelcast to start ...")
+        await sleep(10000)
+    }
+}
+
+start()
```

Comment:
It's better to handle exceptions for root async function invocations like this one and at least log them.